Source/Parallelization/WarpXRegrid.cpp

```diff
@@ -140,14 +144,61 @@ WarpX::LoadBalance ()
 }
 
 
+template <typename MultiFabType> void
+RemakeMultiFab (std::unique_ptr<MultiFabType>& mf, const DistributionMapping& dm,
+                const bool redistribute)
+{
+    if (mf == nullptr) return;
+    const IntVect& ng = mf->nGrowVect();
+    auto pmf = std::make_unique<MultiFabType>(mf->boxArray(), dm, mf->nComp(), ng);
+    if (redistribute) pmf->Redistribute(*mf, 0, 0, mf->nComp(), ng);
+    mf = std::move(pmf);
+}
+
 void
 WarpX::RemakeLevel (int lev, Real /*time*/, const BoxArray& ba, const DistributionMapping& dm)
 {
     if (ba == boxArray(lev))
     {
         if (ParallelDescriptor::NProcs() == 1) return;
 
+        // Fine patch
+        for (int idim=0; idim < 3; ++idim)
+        {
+            RemakeMultiFab(Bfield_fp[lev][idim], dm, true);
+            RemakeMultiFab(Efield_fp[lev][idim], dm, true);
+            RemakeMultiFab(current_fp[lev][idim], dm, false);
+            RemakeMultiFab(current_store[lev][idim], dm, false);
+
+#ifdef AMREX_USE_EB
+            if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::Yee ||
+                WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT ||
+                WarpX::maxwell_solver_id == MaxwellSolverAlgo::CKC){
+                RemakeMultiFab(m_edge_lengths[lev][idim], dm, false);
+                RemakeMultiFab(m_face_areas[lev][idim], dm, false);
+                if(WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT){
+                    RemakeMultiFab(Venl[lev][idim], dm, false);
+                    RemakeMultiFab(m_flag_info_face[lev][idim], dm, false);
+                    RemakeMultiFab(m_flag_ext_face[lev][idim], dm, false);
+                    RemakeMultiFab(m_area_mod[lev][idim], dm, false);
+                    RemakeMultiFab(ECTRhofield[lev][idim], dm, false);
+                }
+
+                m_borrowing[lev][idim] = std::make_unique<amrex::LayoutData<FaceInfoBox>>(amrex::convert(ba, Bfield_fp[lev][idim]->ixType().toIntVect()), dm);
+
+            }
+#endif
+        }
+
```

Comment:
```suggestion

        if(WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT){
            MarkCells();
            ComputeFaceExtensions();
        }
```

Reply:
Should these calls go inside the `#ifdef AMREX_USE_EB` block a little further below?